rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -502,7 +495,16 @@ private boolean allNodesAreAvailable(List<Node> nodes, ResourceManager rm) {
             if (!rm.setNodeAvailable(url).getBooleanValue()) {
                 if (!rm.nodeIsAvailable(url).getBooleanValue()) {
                     logger.info("Node " + url + " removed");
-                    iterator.remove(); // node removed manually by user
+                    // node removed manually by user
+                    // kill the local worker node
+                    String nodeName = null;
+                    try {
+                        nodeName = node.getNodeInformation().getName();
+                        node.getProActiveRuntime().killNode(nodeName);
+                    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)